pkg/multicast/multicast.go

```diff
@@ -374,84 +448,87 @@ func readPackageHeader(r io.Reader) (uint16, uint16, uint32, error) {
 	return n, channelID, seq, nil
 }
 
-func (c *Client) handlePackageHeader(r io.Reader) error {
-	_, channelID, seq, err := readPackageHeader(r)
+func (c *Client) handlePackageHeader(r io.Reader) (offset int, channelID uint16, seq uint32, err error) {
+	_, channelID, seq, err = readPackageHeader(r)
 
 	if err != nil {
 		c.log.Errorw("failed to decode events", "err", err)
-		return err
+		return
 	}
 
 	lastSeq, ok := c.getSeqNum(channelID)
-	if ok && seq != lastSeq+1 { // check for invalid sequence number
-		if seq == 0 {
-			return ErrConnectionReset
-		}
-		return ErrLostPackage
-	}
-	c.setSeqNum(channelID, seq)
-	return nil
+	if !ok {
+		queue := common.NewEventQueue(defaultEventQueueLength)
```

Comment:
Don't need to use `EventQueue` for now, log debug for duplicated package, warning for out of order package, and return `ErrConnectionReset` for package with `seq == 0 and maxUint32 - lastSeq >= 2`